framework/web/UrlRule.php

```diff
@@ -224,6 +230,11 @@ public function init()
         $this->_template = preg_replace('/<([\w._-]+):?([^>]+)?>/', '<$1>', $this->pattern);
         $this->pattern = '#^' . trim(strtr($this->_template, $tr), '/') . '$#u';
 
+        // if host starts with relative scheme, then insert pattern to match any
+        if (strpos($this->host, '//') === 0) {
+            $this->pattern = substr_replace($this->pattern, '[\w]+://', 2, 0);
```

Comment:
@cebe, no, this is correct behavior:

![xdebug](http://erickskrauch.ely.by/screenshot-19-12-2016-at-11-38-32.png)